ledger/src/blocktree.rs

```diff
@@ -1452,9 +1452,23 @@ impl Blocktree {
         }
     }
 
-    pub fn last_root(&self) -> u64 {
+    pub fn last_root(&self) -> Slot {
         *self.last_root.read().unwrap()
     }
+
+    // find the first available slot in blocktree that has some data in it
+    pub fn lowest_slot(&self) -> Slot {
+        for (slot, meta) in self
+            .slot_meta_iterator(0)
+            .expect("unable to iterate over meta")
+        {
+            if slot > 0 && meta.received > 0 {
```

Comment:
It seems we could make the logic of "bounds" sub command of `ledger-tool` to use this for less duplicate code? Also, the `ledger-tool` doesn't have `meta.received > 0` the condition. I dunno which is correct.

From `ledger-tool/src/main.rs`:
>            SubCommand::with_name("bounds")
>            .about("Print lowest and highest non-empty slots. Note: This ignores gaps in slots")

Reply:
I think I want something more than what bounds from ledger tool gives us. Bounds just tells us "this is all lowest and highest slot for which we have some Meta data". In this case, I'm looking for the first non empty slot. Technically they're both similar since we don't update the Mets without Inserting something but that could change. Also I don't need to iterate over the whole ledger. Just need the first item and then exit.

Reply:
Oh, I see. Then, leave the ledger tool logic as it is! Thanks for explaining.

So bounds check description should be changed like this for accuracy?
```patch
-       .about("Print lowest and highest non-empty slots. Note: This ignores gaps in slots")
+       .about("Print lowest and highest slots of metadata in ledger. Note: This ignores gaps in slots")
```

Reply:
Technically true but also not really possible right now. I think we should leave it the way it is.

Also I need to give this lowest_slot logic some more thought. Need to figure out how to avoid the repair bug if someone turns off their node for a few hours and then reboots from a snapshot. The lowest slot will still report the lowest one from the previous run. I might need to persist the boot slot and ignore lowest slots below that.

Reply:
Yeah, that'd be desirable. But just naively doing so will lead just restarting for very a short downtime cuts the lowser slot needlessly. So I think we need some allowance for the exact gap maybe using `snapshot_interval_slots` and the recorded root_slot for the snapshot.

Reply:
Yeah I'm going to give it mood thought tomorrow. Bed time now :)